### PR TITLE
Baseline: Support context file and in-progress issues

### PR DESCRIPTION
**Pull request by psiinon**
Plus a few other tweeks we've found useful when using this script at Mozilla.

### CODE REVIEW

File: build/docker/zap-baseline.py [hunk 1]
```diff
@@ -359,32 +414,35 @@ def main(argv):
...
       while len(alerts) > 0:
+        logging.debug ('Reading ' + str(pg) + ' alerts from ' + str(st))
+        alert_count += len(alerts)
         for alert in alerts:
           plugin_id = alert.get('pluginId')
           if plugin_id in blacklist:
             continue
           if not is_in_scope(plugin_id, alert.get('url')):
             continue
-          if alert.get('risk') == 'Informational':
```
Comment (thc202) [inline build/docker/zap-baseline.py]: Is this being removed on purpose?

Comment (psiinon) [inline build/docker/zap-baseline.py]: No, that was a mistake :/ Added back in now :)

File: build/docker/zap-baseline.py [hunk 2]
```diff
@@ -68,7 +69,10 @@
 # Pscan rules that aren't really relevant, eg example alpha rules
```
Comment (kingthorin) [inline build/docker/zap-baseline.py]: It took me a couple readings to figure out what you meant here, I think for clarity and others not as familiar with the tool an edit might be good:
> eg the examples rules in the alpha set

File: build/docker/zap-baseline.py [hunk 3]
```diff
@@ -86,6 +90,8 @@ def usage():
     print ('    -i                default rules not in the config file to INFO')
     print ('    -j                use the Ajax spider in addition to the traditional one')
     print ('    -l level          minimum level to show: PASS, IGNORE, INFO, WARN or FAIL, use with -s to hide example URLs')
+    print ('    -n context_file   context file which will be loaded prior to spidering the target')
+    print ('    -p progress_file  progress file which specifies issues that are being addressed')
```
Comment (kingthorin) [inline build/docker/zap-baseline.py]: Are you going to add an example file? Or some details here about the format of the file? Reading further it seems like json of some sort but that doesn't really provide enough detail.

File: build/docker/zap-baseline.py [hunk 4]
```diff
@@ -469,18 +536,45 @@ def main(argv):
         with open(base_dir + report_xml, 'w') as f:
           f.write (zap.core.xmlreport())
 
-      print ('FAIL: ' + str(fail_count) + '\tWARN: ' + str(warn_count) + '\tINFO: ' + str(info_count) +  
-        '\tIGNORE: ' + str(ignore_count) + '\tPASS: ' + str(pass_count))
+      print ('FAIL-NEW: ' + str(fail_count) + '\tFAIL-IP: ' + str(fail_inprog_count) + 
+        '\tWARN-NEW: ' + str(warn_count) + '\tWARN-IP: ' + str(warn_inprog_count) +
```
Comment (kingthorin) [inline build/docker/zap-baseline.py]: I'm guessing that "IP" is meant to be 'in progress' but will that be confusing to others? (Need to be careful using IP in an IT context/world 😀 )

File: build/docker/zap-baseline.py (hunk 3 above)
Comment (psiinon) [inline build/docker/zap-baseline.py]: Ok to link to https://github.com/zaproxy/zaproxy/wiki/ZAP-Baseline-Scan and detail them there?

**Comment by psiinon** [comment]
Fixed issues apart from the docs which I'm proposing to put on the wiki

**Comment by kingthorin** [comment]
Sure I'm good with that.

**Comment by psiinon** [comment]
Added dump_log_file(cid) function, use base_dir for progress_file and commented why context_file doesnt use it

**Comment by thc202** [comment]
Thanks!

**Comment by psiinon** [comment]
Have just updated the wiki: https://github.com/zaproxy/zaproxy/wiki/ZAP-Baseline-Scan 
Feedback appreciated :)

**Comment by thc202** [comment]
Looks good to me.

**Pull request merged**